### PR TITLE
Dragging Feature

### PR DESCRIPTION
**Pull request by devzkhalil**
> I'm a die-hard fan of the PHP Debugbar package!

**The reason for this PR!**

In the above image, you can see that I have a menu in the same place as the Debugbar area. Whenever I try to click on the menu, I face issues with the click! I know many people face this problem and may face it in the future!

Therefore, I have tried to implement a feature here that allows the Debugbar to be dragged!

https://github.com/user-attachments/assets/bdbe4f4e-abed-414e-9a78-33388724643a

### CODE REVIEW

**Comment by parallels999** [comment]
I think it should be more integrated into the debugbar

File: src/DebugBar/Resources/debugbar.js [hunk 1]
```diff
@@ -1293,4 +1293,88 @@ if (typeof(PhpDebugBar) == 'undefined') {
 
     });
 
+    // ============= Handling Dragging Behavior ============
+    var storedPosX = localStorage.getItem('phpdebugbarPositionX') || 0;
+    var screenWidth = $(window).width();
+
+    $(document).on('click', '.phpdebugbar', function () {
+        let isClosed = $('.phpdebugbar').hasClass('phpdebugbar-closed');
+        if (!isClosed) {
```
Comment (parallels999) [inline src/DebugBar/Resources/debugbar.js:1302]: `isClosed` method exists
https://github.com/maximebf/php-debugbar/blob/c9cb28d5ee6f657fcc48440f38db8bbd140bef83/src/DebugBar/Resources/debugbar.js#L840-L842

File: src/DebugBar/Resources/debugbar.js [hunk 2]
```diff
@@ -1293,4 +1293,88 @@ if (typeof(PhpDebugBar) == 'undefined') {
...
+            // Save current position to localStorage when closing the debugbar
+            storedPosX = localStorage.getItem('phpdebugbarPositionX');
```
Comment (parallels999) [inline src/DebugBar/Resources/debugbar.js:1304]: maybe more consistent with the other storage keys
like `phpdebugbar-restore-position`
https://github.com/maximebf/php-debugbar/blob/c9cb28d5ee6f657fcc48440f38db8bbd140bef83/src/DebugBar/Resources/debugbar.js#L1151
https://github.com/maximebf/php-debugbar/blob/c9cb28d5ee6f657fcc48440f38db8bbd140bef83/src/DebugBar/Resources/debugbar.js#L790-L791

File: src/DebugBar/Resources/debugbar.js [hunk 3]
```diff
@@ -1293,4 +1293,88 @@ if (typeof(PhpDebugBar) == 'undefined') {
...
+        }
+    });
+
+    $(document).ready(function () {
+        // Check local storage for saved position
+        var savedPosX = localStorage.getItem('phpdebugbarPositionX');
+        var phpdebugbarIsOpen = localStorage.getItem('phpdebugbar-open');
+
+        if (savedPosX !== null && phpdebugbarIsOpen == 0 && screenWidth > savedPosX) {
+            $('.phpdebugbar').css('left', savedPosX + 'px');
```
Comment (parallels999) [inline src/DebugBar/Resources/debugbar.js:1319]: Calling to search the entire document for classes like `.phpdebugbar` would be avoided if the class properties are reused, like `this.$el.css('left', savedPosX + 'px')`
https://github.com/maximebf/php-debugbar/blob/c9cb28d5ee6f657fcc48440f38db8bbd140bef83/src/DebugBar/Resources/debugbar.js#L488

File: src/DebugBar/Resources/debugbar.js [hunk 4]
```diff
@@ -1293,4 +1293,88 @@ if (typeof(PhpDebugBar) == 'undefined') {
...
+            storedPosX = localStorage.getItem('phpdebugbarPositionX');
+            $('.phpdebugbar').css('left', '0px');
+        } else {
+            // Restore saved position from localStorage when opening the debugbar
+            localStorage.setItem('phpdebugbarPositionX', storedPosX);
+            $('.phpdebugbar').css('left', storedPosX + 'px');
+        }
+    });
+
+    $(document).ready(function () {
```
Comment (parallels999) [inline src/DebugBar/Resources/debugbar.js:1313]: instead of waiting for jquery to load, it should be executed in the debugbar render

**Comment by erikn69** [comment]
@devzkhalil hi, did you need help??

**Comment by erikn69** [comment]
@devzkhalil hi, 
I found some bugs, I have fix them on https://github.com/devzkhalil/php-debugbar/pull/1, 
I also reorganized your code

**Comment by devzkhalil** [comment]
Great changes @erikn69

**Comment by parallels999** [comment]
looks good to me, but  I'm not a maintainer, @barryvdh must review it

**Review by erikn69 — APPROVED** [review]

File: src/DebugBar/Resources/debugbar.js [hunk 5]
```diff
@@ -485,6 +491,7 @@ if (typeof(PhpDebugBar) == 'undefined') {
             }
 
             var self = this;
+            this.$el.css('display', 'block');
```
Comment (erikn69) [inline src/DebugBar/Resources/debugbar.js:494]: this affects print mode, and in iframes
fixed on https://github.com/devzkhalil/php-debugbar/pull/3

**Review by parallels999 — APPROVED** [review]

**Comment by angeljqv** [comment]
@barryvdh hi, it happens, look:
image

**Comment by devzkhalil** [comment]
@angeljqv 
Those are your app contents behind the Laravel Debugbar icon, am i right brother?

**Comment by angeljqv** [comment]
Yes

**Comment by devzkhalil** [comment]
Thank you @angeljqv  for sharing your scenario.
 It may help others facing similar situations.

**Review by angeljqv — APPROVED** [review]

**Review by atonies — APPROVED** [review]

**Review by PaolaRuby — APPROVED** [review]

**Comment by parallels999** [comment]
@barryvdh hi, any review??

**Comment by barryvdh** [comment]
I don't really have this issue myself but I can see it being useful.  Will try it out.

**Comment by barryvdh** [comment]
So tried it, a few things come to mind:
 - the left border for the icon is missing
 - would it make sense to 'snap' to the right? If I drag it to the right corner, I would expect it to stick, even if I make it larger. It does work when making it smaller, but not larger. Not sure what the cleanest is, but I guess if we could 'flip' the offset from left to right or something?

**Comment by barryvdh** [comment]
What do you think about snapping to the right? And maybe the middle?

**Comment by barryvdh** [comment]
And I'm getting some misfires for extending, because it probably thinks it's dragging.

**Comment by devzkhalil** [comment]
> What do you think about snapping to the right? And maybe the middle?


I don't think anyone will need to drag it after opening the Debugbar content area/larger.
Still, I’m trying to show you something! I can show you something very quickly!

**Comment by devzkhalil** [comment]
> And I'm getting some misfires for extending, because it probably thinks it's dragging.

ups!
I will check and fix.......

**Comment by devzkhalil** [comment]
> So tried it, a few things come to mind:
> 
> * the left border for the icon is missing
> * would it make sense to 'snap' to the right? If I drag it to the right corner, I would expect it to stick, even if I make it larger. It does work when making it smaller, but not larger. Not sure what the cleanest is, but I guess if we could 'flip' the offset from left to right or something?

Something like this for larger?

https://github.com/user-attachments/assets/07f32a32-49dc-4d46-af50-22683f21a33b

**Comment by barryvdh** [comment]
Eh no sorry, thats not really what I meant. I just meant that if I drag the button to the right corner and then make the screen larger, I would probably expect the button to stick to the corner, instead of being half way.
With sticky/snapping I would mean that if you get within a few pixels of the corner, it woud 'snap' into place in the corner.

**Comment by barryvdh** [comment]
This starting to look nice, thanks :)

Few things I notice
 - Sometimes debugbar doesn't open on first click when closed. So might be some incorrect state somewhere?
 - I see you change the positioning to a percentage. I'm not 100% sure what makes the most sense yet. When positioning just outside the corner, I suspect you would want to keep it fixed. If it's more to the center, percentage might be more useful (although that seems like an odd place). But not really sure what makes sense.

**Comment by barryvdh** [comment]
See #697 
https://github.com/user-attachments/assets/6cb0d076-4c3f-4812-a141-28e4f28463ac

**Comment by barryvdh** [comment]
Soo I played with it a bit, but not 100% sure. See https://github.com/php-debugbar/php-debugbar/pull/701

> I tested this, and I'm not 100% sure about it. I like the idea but it feels really like and edge case. I can see the appeal of moving the debugbar to the right instead of left, but I think that should be either configurable in the UI or as option.
> The dragging has a bit of a downside of keeping the draggable state when switching tabs or something. Not really sure but it doesn't feel reliable yet.

And to be honest, in the 10+ years of using this, I don't I've heard this feature request yet, so not really sure how necessary this is.

If other feel like they also need this, please let me know.

**Pull request closed without merging**